compiler/rustc_driver_impl/src/lib.rs

```diff
@@ -1307,15 +1310,14 @@ pub fn catch_with_exit_code(f: impl FnOnce() -> interface::Result<()>) -> i32 {
 static ICE_PATH: OnceLock<Option<PathBuf>> = OnceLock::new();
 
 fn ice_path() -> &'static Option<PathBuf> {
+    ice_path_with_config(None)
+}
+
+fn ice_path_with_config(config: Option<&UnstableOptions>) -> &'static Option<PathBuf> {
```

Comment:
It would also perhaps be useful if we could do something like:

```
if ICE_PATH.get().is_some() && config.is_some() && cfg!(debug_assertions) {
    warn!("ICE_PATH has already been initialized -- files may be emitted at unintended paths")
}
```